pkgs/by-name/tw/twitch-hls-client/package.nix

```diff
@@ -0,0 +1,34 @@
+{
+  lib,
+  rustPlatform,
+  fetchFromGitHub,
+  stdenv,
+  darwin,
+  ...
+}:
+
+rustPlatform.buildRustPackage {
+  pname = "twitch-hls-client";
+  version = "unstable-2024-10-16";
+
+  src = fetchFromGitHub {
+    owner = "2bc4";
+    repo = "twitch-hls-client";
+    rev = "4d1b6fe71d18f7c7dbceea04a99c98db16cd0d0b";
+    hash = "sha256-ja2pFmJNFfNMaNlkgvfxCFYyuVup6CgIEhnbzyk/kro=";
+  };
+
+  cargoHash = "sha256-XL/LSEhWZB7p9wgisKcGWDknEvNNFje5WLQazS1730o=";
+
+  buildInputs = lib.optionals stdenv.isDarwin [
+    darwin.apple_sdk.frameworks.Security
+  ];
```

Comment:
There is a issue that package maintainer's doesn't think this package should be support in Darwin. I'm not sure if this operation is correct.

https://github.com/2bc4/twitch-hls-client/blob/4d1b6fe71d18f7c7dbceea04a99c98db16cd0d0b/nix/default.nix#L48

And by the ways, consider that the package maintainer is already packing itself, should this still be contains in nixpkgs?  @wrvsrx

Reply:
> There is a issue that package maintainer's doesn't think this package should be support in Darwin. I'm not sure if this operation is correct.
> 
> https://github.com/2bc4/twitch-hls-client/blob/4d1b6fe71d18f7c7dbceea04a99c98db16cd0d0b/nix/default.nix#L48

Nix package and flakes were brought to the repository by a single-time contributor, not by the maintainer himself, so it looks like the PR author just added support for the platform he uses that package on (Linux).

 Also, the real package maintainer states platforms in README clearly showing macOS as well as windows so that little CLI tool could be considered cross-platform: 
 https://github.com/2bc4/twitch-hls-client/blob/4d1b6fe71d18f7c7dbceea04a99c98db16cd0d0b/README.md?plain=1#L76

Reply:
> And by the ways, consider that the package maintainer is already packing itself, should this still be contains in nixpkgs? @wrvsrx

Also, I don't think there are any analogues available that allow both watching and recording twitch from the CLI, so I can't see the reason for leaving this package outside of nixpkgs and thus depriving users of another useful tool.

Reply:
sor, maybe there.is my mistake. 
i agree with your opinion.
And the above part of 
（if software maintainer already pakcing there package,should it still contain in nixpkgs）just one of my question, about which package should be contains in nixpkgs , not a comment on your package

Reply:
I'm not a maintainer of the package. Maybe you want to @ another person?